expression/builtin_time_vec.go

```diff
@@ -790,11 +790,29 @@ func (b *builtinDateDiffSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column
 }
 
 func (b *builtinCurrentDateSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCurrentDateSig) vecEvalTime(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	nowTs, err := getStmtTimestamp(b.ctx)
+	if err != nil {
+		return err
+	}
+
+	tz := b.ctx.GetSessionVars().Location()
+	year, month, day := nowTs.In(tz).Date()
+	mysqlTm := types.FromDate(year, int(month), day, 0, 0, 0, 0)
+
+	n := input.NumRows()
+	result.ResizeTime(n, false)
+	times := result.Times()
+	for i := 0; i < n; i++ {
+		times[i] = types.Time{
```

Comment:
would it be faster if we move the object allocation outside the for loop, i.e:
```go
curDate := types.Time{
	Time: mysqlTm,
	Type: mysql.TypeDate,
	Fsp:  0,
}
for i := 0; i < n; i++ {
	// copy curDate to the vector.
}
```

Reply:
Sure, will fix in a moment.

Reply:
Before:

```
BenchmarkVectorizedBuiltinTimeFunc/builtinCurrentDateSig-VecBuiltinFunc-12         	  195573	      6102 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinCurrentDateSig-NonVecBuiltinFunc-12      	    6634	    182992 ns/op	       0 B/op	       0 allocs/op
```

After:

```
BenchmarkVectorizedBuiltinTimeFunc/builtinCurrentDateSig-VecBuiltinFunc-12         	  987668	      1197 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinCurrentDateSig-NonVecBuiltinFunc-12      	    7215	    172915 ns/op	       0 B/op	       0 allocs/op
```